{"url":"https://api.github.com/repos/BeamNG/BeamNGpy/issues/133","repository_url":"https://api.github.com/repos/BeamNG/BeamNGpy","labels_url":"https://api.github.com/repos/BeamNG/BeamNGpy/issues/133/labels{/name}","comments_url":"https://api.github.com/repos/BeamNG/BeamNGpy/issues/133/comments","events_url":"https://api.github.com/repos/BeamNG/BeamNGpy/issues/133/events","html_url":"https://github.com/BeamNG/BeamNGpy/issues/133","id":946188263,"node_id":"MDU6SXNzdWU5NDYxODgyNjM=","number":133,"title":"Accessing to BeamNG.tech from remote host","user":{"login":"beunguk","id":3955650,"node_id":"MDQ6VXNlcjM5NTU2NTA=","avatar_url":"https://avatars.githubusercontent.com/u/3955650?v=4","gravatar_id":"","url":"https://api.github.com/users/beunguk","html_url":"https://github.com/beunguk","followers_url":"https://api.github.com/users/beunguk/followers","following_url":"https://api.github.com/users/beunguk/following{/other_user}","gists_url":"https://api.github.com/users/beunguk/gists{/gist_id}","starred_url":"https://api.github.com/users/beunguk/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/beunguk/subscriptions","organizations_url":"https://api.github.com/users/beunguk/orgs","repos_url":"https://api.github.com/users/beunguk/repos","events_url":"https://api.github.com/users/beunguk/events{/privacy}","received_events_url":"https://api.github.com/users/beunguk/received_events","type":"User","site_admin":false},"labels":[],"state":"open","locked":false,"assignee":{"login":"Palculator","id":37553150,"node_id":"MDQ6VXNlcjM3NTUzMTUw","avatar_url":"https://avatars.githubusercontent.com/u/37553150?v=4","gravatar_id":"","url":"https://api.github.com/users/Palculator","html_url":"https://github.com/Palculator","followers_url":"https://api.github.com/users/Palculator/followers","following_url":"https://api.github.com/users/Palculator/following{/other_user}","gists_url":"https://api.github.com/users/Palculator/gists{/gist_id}","starred_url":"https://api.github.com/users/Palculator/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Palculator/subscriptions","organizations_url":"https://api.github.com/users/Palculator/orgs","repos_url":"https://api.github.com/users/Palculator/repos","events_url":"https://api.github.com/users/Palculator/events{/privacy}","received_events_url":"https://api.github.com/users/Palculator/received_events","type":"User","site_admin":false},"assignees":[{"login":"Palculator","id":37553150,"node_id":"MDQ6VXNlcjM3NTUzMTUw","avatar_url":"https://avatars.githubusercontent.com/u/37553150?v=4","gravatar_id":"","url":"https://api.github.com/users/Palculator","html_url":"https://github.com/Palculator","followers_url":"https://api.github.com/users/Palculator/followers","following_url":"https://api.github.com/users/Palculator/following{/other_user}","gists_url":"https://api.github.com/users/Palculator/gists{/gist_id}","starred_url":"https://api.github.com/users/Palculator/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Palculator/subscriptions","organizations_url":"https://api.github.com/users/Palculator/orgs","repos_url":"https://api.github.com/users/Palculator/repos","events_url":"https://api.github.com/users/Palculator/events{/privacy}","received_events_url":"https://api.github.com/users/Palculator/received_events","type":"User","site_admin":false}],"milestone":null,"comments":5,"created_at":"2021-07-16T11:09:00Z","updated_at":"2021-08-09T06:52:26Z","closed_at":null,"author_association":"NONE","active_lock_reason":null,"body":"Hi, \r\n\r\nCurrently, I have two machines that one is Windows10 and Ubuntu 18.04.\r\nI installed BeamNG.tech v0.21.3 and BeamNGpy==0.19.1 on the windows machine, and I installed BeamNGpy==0.19.1 on Ubuntu.\r\nIf I understand correctly, BeamNG.tech supports multiple clients and I want to access BeamNG.tech from the Ubuntu machine.\r\n\r\nSo what I did is that I launched BeanNG.tech from the Windows machine. Then try to connect BeanNG.tech from the Ubuntu machine. On the BeamNG.tech console, it seems to accept the connection, but it closes or raises an error (timeout) after sending hello() or other msgs to BeamNG.tech.\r\n\r\nSo basically, it's like:\r\n\r\nOn windows, this is what I did, and it works. It launches BeamNG.tech with the console.\r\n```python\r\nfrom beamngpy import BeamNGpy, Scenario, Vehicle\r\n# Instantiate BeamNGpy instance running the simulator from the given path,\r\n# communicating over localhost:64256\r\nbng = BeamNGpy('localhost', 64256)\r\n# Launch BeamNG.tech\r\nbng.open()\r\n```\r\n\r\nOn Ubuntu, it works until bng.open(launch=False, deploy=False) , but at the hello() method, connection hangs.\r\nAnd I modified some part of to skip checking binary file (.exe) existence in BeamNGpy package.\r\nfrom beamngpy import BeamNGpy, Scenario, Vehicle\r\n```python\r\n# Instantiate BeamNGpy instance running the simulator from the given path,\r\n# communicating over 'ip to the windows machine':64256\r\nbng = BeamNGpy('ip to the windows machine', 64256)\r\n# Launch BeamNG.tech\r\nbng.open(launch=False, deploy=False) # it returns a BeamNGpy object as expected.\r\nbng.hello()\r\n```\r\n\r\nWhen I am trying the above script, I got these logs from the console. (I also disabled all firewalls on windows and ubuntu.)\r\n```\r\n2806.073|I|GELua.util_researchGE.ResearchGE|Accepted new client: {Ubuntu Machine IP}/64256\r\n2807.950|E|GELua.ResearchCom|Error reading from socket: timeout\r\n2807.954|E|GELua.ResearchCom|Error reading from socket: tcp{client}: 0000020DCFB70500 - timeout\r\n```\r\n\r\nI know the setting is a bit complicated, but my development runs on a linux based system.\r\nSo I really want to access and control the simulator through the linux based system. Is there something I missed?","closed_by":null,"reactions":{"url":"https://api.github.com/repos/BeamNG/BeamNGpy/issues/133/reactions","total_count":0,"+1":0,"-1":0,"laugh":0,"hooray":0,"confused":0,"heart":0,"rocket":0,"eyes":0},"timeline_url":"https://api.github.com/repos/BeamNG/BeamNGpy/issues/133/timeline","performed_via_github_app":null,"state_reason":null}